editor/index.js

```diff
@@ -12,6 +19,10 @@ import './assets/stylesheets/main.scss';
 import Layout from './layout';
 import { createReduxStore } from './state';
 
+// Configure moment globally
+moment.locale( settings.l10n.locale );
```

Comment:
It's independent in `wp-date` but an app like Gutenberg can make it global explicitly. That's my thinking. But I don't mind moving this to `wp-date` but we'd have to acknowledge that `wp-date` is moment dependent (externally).